src/mainboard/starfive/visionfive2/bt0/src/main.rs

```diff
@@ -80,10 +69,11 @@ pub unsafe extern "C" fn start() -> ! {
         "csrw   mtvec, zero",
         // 1. suspend non-boot hart
         // hart 0 is the S7 monitor core; 1-4 are U7 cores
-        "li     a1, 0",
+        "li     a1, 1",
         "csrr   a0, mhartid",
         "bne    a0, a1, .nonboothart",
         // 2. prepare stack
+        // FIXME: each hart needs its own stack
```

Comment:
Since we are loading a payload, isn't the payload responsible for creating its runtime stacks?

Also, we are only running in the supervisor core with a `riscv64imac` target. All the other cores are running under the `riscv64gc` target (a superset of `riscv64imac`). I'm not sure about the safety of doing any setup that crosses that boundary. Thoughts?

Reply:
To clarify: _supervisor_ is a mode in RISC-V; a core is called a _hart_ - and the SoC starts each hart in machine mode.
I think you mean _monitor core_, which I think is what they call hart 0 in the manual and the [data sheet](https://doc-en.rvspace.org/JH7110/PDF/JH7110_Datasheet.pdf), an S7 (RV64IMAC) that does not have the supervisor mode / has no MMU, since it's an [embedded core](https://www.sifive.com/blog/the-sifive-20g1-update-for-7-series-core-ip) (not that well documented...).
That is why it cannot run the same Linux kernel as the others and why we need to switch to running on harts 1-4, while one of them needs to be chosen to do the initialization that we do in oreboot and has control over the peripherals, e.g. for serial output. That is why this PR includes the commit to switch to running on hart 1.
Via the optional SBI that we want to offer you would drop into S-mode. That is what #694 is about.

The compile target means that we use only that set (subset) of instructions; that has no impact on the cores, they only need to support the respective set of used instructions.

Reply:
> The compile target means that we use only that set (subset) of instructions; that has no impact on the cores, they only need to support the respective set of used instructions.

Right, I guess it would only matter going in the opposite direction (GC to IMAC), since the former includes floating point instructions, and the latter doesn't.

Anyway, thanks for the detailed response.